### PR TITLE
feat(Dispatcher): add support for other event than click #3

### LINKED ISSUE #3: dispatcher should support other events than the click

### PR DESCRIPTION
**Pull request by acateland**
Enhance the Dispatcher component so it can now inject a dispatchable actionCreator for any event supported by react or custom event required that it start by 'on' followed by at least on caracter.

breaking change :
dispatcher doesn't use anymore action prop, but any on(eventName) property
mapDispatchToProps api now take required props object parameter to

Fixes #3

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,7 +55,7 @@
     "react": "^15.3.1",
     "react-dom": "^15.3.1",
     "react-redux": "^4.4.5",
-    "react-router": "^2.6.1",
+    "react-router": "3.0.0-alpha.3",
```
Comment (jmfrancois) [inline package.json]: i don't think it's a wanted change here.

File: __tests__/action.test.js [hunk 2]
```diff
@@ -120,36 +120,33 @@ describe('uiAbstraction action', () => {
...
+    expect(typeof mapped.onClick).toBe('function');
+    expect(typeof mapped.onAnything).toBe('function');
   });
-  it('mapDispatchToProps should throw exception if no action found', () => {
-    const dispatched = {
-      called: false,
-    };
+  it('mapDispatchToProps should merge props with any resolved dispatchable action creator', () => {
     function dispatch(action) {
       dispatched.called = true;
```
Comment (jmfrancois) [inline __tests__/action.test.js]: dispatched is not defined anymore. I think we should use jest.fn to defined the dispatch function

File: src/Dispatcher.js [hunk 3]
```diff
@@ -1,41 +1,53 @@
...
+
 Dispatcher.propTypes = {
-  action: React.PropTypes.oneOfType([
-    React.PropTypes.string,
-    React.PropTypes.object,
-  ]),
-  model: React.PropTypes.object,
-  onClick: React.PropTypes.func,
-  children: React.PropTypes.object,
+  children: PropTypes.node.isRequired,
```
Comment (jmfrancois) [inline src/Dispatcher.js]: How could you test the propTypes dynamicly ? I guess we may add all events supported by react ?

**Comment by jmfrancois** [comment]
related issue #3

**Comment by talend-glorieux** [comment]
LGTM

File: package.json (hunk 1 above)
Comment (acateland) [inline package.json]: react-router add a feature who we will need to use to handle dynamic url

File: src/Dispatcher.js (hunk 3 above)
Comment (acateland) [inline src/Dispatcher.js]: this is by design to support an y custom event we would like to support, adding this to PR description

File: __tests__/action.test.js (hunk 2 above)
Comment (acateland) [inline __tests__/action.test.js]: agree we should move the dispatching test to the Dispatcher components tests.

**Pull request merged**